salesforce/salesforce-inbound-endpoint/salesforce-inbound-endpoint-1.0.0/src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -0,0 +1,342 @@
+        if (log.isDebugEnabled()) {
+            log.debug("Starting to load the salesforce credentials");
+        }
+        if (properties.getProperty(SalesforceConstant.CONNECTION_TIMEOUT) == null) {
+            connectionTimeout = SalesforceConstant.CONNECTION_TIMEOUT_DEFAULT;
+        } else {
+            try {
+                connectionTimeout = Integer.parseInt(properties.getProperty(SalesforceConstant.CONNECTION_TIMEOUT));
+            } catch (NumberFormatException e) {
+                log.error("The Value should be in Number", e);
+            }
+        }
+
+        if (properties.getProperty(SalesforceConstant.READ_TIMEOUT) == null) {
+            readTimeout = SalesforceConstant.READ_TIMEOUT_DEFAULT;
+        } else {
+            try {
+                readTimeout = Integer.parseInt(properties.getProperty(SalesforceConstant.READ_TIMEOUT));
+            } catch (NumberFormatException e) {
+                log.error("The Value should be in Number", e);
+            }
+        }
+
+        if (properties.getProperty(SalesforceConstant.WAIT_TIME) == null) {
+            connectionTimeout = SalesforceConstant.WAIT_TIME_DEFAULT;
+        } else {
+            try {
+                waitTime = Integer.parseInt(properties.getProperty(SalesforceConstant.WAIT_TIME));
+            } catch (NumberFormatException e) {
+                log.error("The Value should be in Number", e);
+            }
+        }
+    }
+
+    public Object poll() {
+        //Establishing connection with Salesforce streaming api.
+        try {
+            if (!isPolled) {
+                makeConnect();
+                isPolled = true;
+            }
+        } catch (Exception se) {
+            handleException("Error while setup the Salesforce connection.", se);
+        }
+        return null;
+    }
+
+    /**
+     * Injecting the salesforce Stream messages to the ESB sequence.
+     *
+     * @param message the salesforce response status
+     */
+    private void injectSalesforceMessage(Message message) {
+        if (injectingSeq != null) {
+            injectMessage(message.getJSON(), SalesforceConstant.CONTENT_TYPE);
+            if (log.isDebugEnabled()) {
+                log.debug("injecting salesforce message to the sequence : " + injectingSeq);
+            }
+        } else {
+            handleException("the Sequence is not found");
+        }
+    }
+
+    private void handleException(String msg, Exception ex) {
+        log.error(msg, ex);
+        throw new SynapseException(ex);
+    }
+
+    private void handleException(String msg) {
+        log.error(msg);
+        throw new SynapseException(msg);
+    }
+
+    public void destroy() {
+        try {
+            if (client != null) {
+                client.disconnect();
+                if (log.isDebugEnabled()) {
+                    log.debug("The Salesforce stream has been shutdown !");
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Fixed in https://github.com/wso2-attic/esb-connectors/pull/557/commits/c9ffbe2fa8d7cd7e3c7517aba0ad656a291a1350